### ISSUE TITLE
(More pt. 2 pt II) UI-redesign issues

### ISSUE THREAD

**Issue by Tiggles** [SDU-eScience/UCloud#4141]
### Essential

- [x] Shortcuts are only shown for FileBrowse
- [x] Allow user to hide shortcuts, regardless of window width
- [x] SvgCaches are cleared on ResourceBrowser unmount.
- [x] ResourceBrowser deselects rows when dragging outside component.
- [x] Rename "Create SSH key" to "Add SSH key"
- [x] Wrap "Add SSH key" in main-container
- [x] Rename "SSH key" to "Add SSH key" on Add SSH key page
- [x] [addProjectListener makes multiple `open`-calls](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2060447505)
- [x] External links are not remapped in Markdown previews (.md files are considered text/code)
- [x] ProductSelector position is off for `Drives` until a resize happens
- [x] ProductSelector for public links is at the bottom
- [x] "Submit application" button moves to the right when opening the sidebar on hover
- [x] Not enough stat3Width for ProjectInvite
- [x] "Shared by" on "Invites" card on dashboard appears on two lines (Also Project Invites)
- [x] [Ellipses are added, even when no folder is present. (Only in one folder in the drive, in the image)](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2049576303)
- [x] HTMLTooptip can go below screen height
- [x] Syncthing FileBrowser crash on navigation - Happens on hovering over Tooltip before navigating
- [x] Sidebar navigation crash - (How to reproduce?)
- [x] Simplify ThemeToggle 
- [x] [Drive-dropdown in FileBrowse makes navbar too wide](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2044653842)
- [x] #4168
- [x] [Empty resource creation](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2017515148)
- [x] [Search query issue](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2018032455)
- [x] [Show shortcuts for embedded, with enough space](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2022645904)
- [x] Dashboard listrows should all match resourcebrowser row heights
- [x] Inconsistent padding/margin for operations (See Drive-browse and File-browse)
- [x] Fix missing "Only show member files" when going from My workspace to project
- [x] Center Grant Application page content
- [x] Make search bar twice as big (Resource browser and App search)
- [x] [Grant application header styling](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2036779499)
- [x] Read/Edit icons from Shares need right margin/padding
- [x] Revert to what was clicked when stopping hovering
- [x] Stats are being pushed by long names (Drives, for instance)
- [x] Syncthing icon alignment on filerow
- [x] Use different colors for shares in sidebar (re-use share folder in FileBrowse)
- [x] `user-select`/`-webkit-user-select` should have `none` set in more places
- [x] Remove text "Resources" from workspaces
- [x] Dead link favorites can't be removed
- [x] Remove headers for link from "Resources" secondary sidebar
- [x] Every secondary sidebar should start at Files icon y-top in primary sidebar
- [x] Rename "Categories" to "Store" in secondary sidebar
- [x] Remove "Dashboard" and "Applications" text from Dashboard and App store
- [x] Invites on Dashboard: Date position is inconsistent between Projects and Shares
- [x] Invites on Dashboard: Tooltip should be added for Avatar
- [x] Invites on Dashboard: Show avatar instead of name
- [x] Show which links are clickable on dashboard, with icon
- [x] DeiC z-index on Dashboard is too dang high
- [x] Dashboard product shows product with wrong monospace font
- [x] Runs in progress should be green
- [x] Providers and products rows should match in height
- [x] Center stats1 and stat2. Right-align stat3
- [x] Remove "Sensitivity" header
- [x] Add tooltip to sensitivity
- [x] Fixed size for stats1..3
- [x] Shortcut styling should match styling on cheatsheet
- [x] Omit shortcuts on smaller screens
- [x] Only show favorite star on hover OR favorite
- [x] Icon sizes should be 30px
- [x] "Manage synchronization" -> "Sync"
- [x] Close context switcher on window scroll
- [x] Input inconsistencies - Font features, margins kerning, maybe
- [x] The blue line under Job Runs should extend all the way, if only possibility
- [x] Search animation on resource browser is missing
- [x] Wobbly icon on Refresh hover (at least on app store) [Edit: Can't reproduce]
- [x] Layout needs to be more consistent (???)
- [x] Move sidebar sticky-button to bottom
- [x] Fix Safari wanting to download App Interfaces instead of downloading them
- [x] Notifications-window doesn't close if open and clicking notification icon
- [x] Sidebar "Sub-projects" goes to "Allocations"(Leaving as is)
- [x] Create public link in modal adds line at the bottom that can't be properly interacted with
- [x] "Info" icon is not centered
- [x] https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-1976291371
- [x] https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-1976298934
- [x] https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-1976307238
- [x] [Product selector text is not centered (edit: the table header overlaps with first element)](https://github.com/SDU-eScience/UCloud/issues/4141#issuecomment-1976524879)
- [x] Members-UI: "Remove" button is not disabled with tooltip for PI
- [x] Members-UI: Avatars are not re-fetched on project change
- [x] Members-UI: "Remove button" is shown for if user in project
- [x] Members-UI: No re-direct if removing self from project.
- [x] (old bug) Members-UI: Navigating to `/app/projects/members` with no active project shows blank page instead of redirecting to Dashboard
- [x] Members-UI: Hide or disable "New group" button for users
- [x] Members-UI: Hide move-to-group button for users
- [x] Members-UI: Hide "Remove" button for users
- [x] Members-UI: Add missing "Transfer PI" functionality
- [x] Styling for inline creation inputs in modals is off. See for instance FileBrowse modal and click "Create folder".
- [x] Inline creation for modals doesn't work properly
- [x] Size for .title in ResourceBrowser-rows needs to be equal to full rowsize, minus statRowWidth 1 through 3, and eventual favorite icon and resource icon
- [x] Filter dropdown shortcuts should be right aligned
- [x] Don't group "Upload files" and "Create folder"
- [x] [The add-to-hours buttons for jobs don't do what they say. Make the behaviour clearer.](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-1988000221)
- [x] [Job selector column titles width is off](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-1993763641)
- [x] SKUs page styling is off (Missing hover effect. Missing pointer-cursor. Styling for modal is wrong)
- [x] Next step/Previous step and modal heights for Syncthing for "Add device"-wizard
- [x] Allow deletion of multiple drives
- [x] [Confirm buttons in operations list has wrongly sized SVGs](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2014966362)
- [x] Confirm buttons' checkmark is red on completion
- [x] User-select in dropdowns should not be here
- [x] Missing License-server placeholder text
- [x] Wrong placeholder color on login
- [x] Syncthing warning does not replace the drive id with the pretty name
- [x] [Handle unknown apps](https://github.com/SDU-eScience/UCloud/issues/4141#issuecomment-2039217526)
- [x] Revert max-height for contextmenu
- [x] [Selector width sizing issue on Chrome](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2042088692)
- [x] [Rename drive input field overlaps with provider icon](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2042154787)
- [x] [Delete Drive button lost its color in contextmenu](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2042175818)
- [x] FileBrowse: Readonly folders have drag-to-select disabled due to text position of `(read-only)`.
- [x] Move (React-implemented) tooltip to right of cursor with small offset.
- [x] "Show hidden files" missing from FileBrowse
- [x] Breadcrumbs is too eager to truncate paths, even with enough width
- [x] Creating nested paths shows the final folder as the name, instead of the first. (e.g. a/b/c will show c in the current folder, but should show a)

### In the middle

- [x] For multi-node jobs, be clear about which nodes you open the terminal or interface to

### Minor

- [x] Find use for provider icons next to navbar? (Navigates to drives for now)
- [ ] Generally remove filters for resources
- [x] [Favorite icon flicker](https://github.com/sdu-eScience/uCloud/issues/4141#issuecomment-2011658977)

**Comment by tedyted26** [comment]
<img width="1520" alt="Screenshot 2024-03-04 at 11 02 37" src="https://github.com/SDU-eScience/UCloud/assets/61278272/3c67a5b8-29b4-4911-bef0-3689b788ba00">
Creating a folder from the add folder menu when configuring a job, then clicking "use" leaves this on the job parameters

**Comment by tedyted26** [comment]
Clicking on "Manage syncronization" from modal in any job parameter, and then going back, resets all my job parameters

**Comment by Tiggles** [comment]
<img width="921" alt="Screenshot 2024-03-04 at 13 55 42" src="https://github.com/SDU-eScience/UCloud/assets/7610257/3226a2f7-6e28-45d4-801a-53eb6e618214">

**Comment by xiroV** [comment]
_Empty trash_ modal needs restyling. Could be resized, and doesn't add the background overlay. Also I don't think we use the underline of modal-headings anymore.

Empty-ing of trash redirects to `/drives` page. Would expect either no redirect, or redirect to the root of the drive that the trash folder belongs to.

**Comment by Tiggles** [comment]
> <img alt="image" width="1616" src="https://private-user-images.githubusercontent.com/1782437/310840559-551d5881-7829-4a7d-8fb5-ff65386dc3e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4MDcyMDgsIm5iZiI6MTcwOTgwNjkwOCwicGF0aCI6Ii8xNzgyNDM3LzMxMDg0MDU1OS01NTFkNTg4MS03ODI5LTRhN2QtOGZiNS1mZjY1Mzg2ZGMzZTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDdUMTAyMTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2NlMDkzNWExMTNkNTdmYjZlYzdlNTIzZjM1NjI5MDIyNTE4YTczYTI5OWVlMTllMjc0OGZlNzA2MTlmMjRhYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ._kueu_lVH_kdTosQscVpyegLXbZGJjXMy_JMju2b6Kw">

Should have been fixed in d4f2dfd149865521bb61263cea1b0b89282c1a86

**Comment by tedyted26** [comment]
(Not sure if its an issue or a design choice)
In job parameters, when using the buttons to add hours, if the current selected hours are less than the hours the button adds, the number does not add up, but changes to the one on the button. For example, having 1h and pressing +8 will result in 8h, not 9. Having 8h and pressing +24 will result in 24h, not 32.

**Comment by Tiggles** [comment]
It's a design choice, but I'm not sure we're done tuning it yet.

**Comment by tedyted26** [comment]
<img width="1680" alt="Screenshot 2024-03-11 at 10 54 40" src="https://github.com/SDU-eScience/UCloud/assets/61278272/e127833d-a82e-49c1-bdb7-e794f293ee68">
Shape (shadow?) of hold to confirm icon

**Comment by DanThrane** [comment]
It is not possible to create a folder when selecting an input folder for an app. The enter key does nothing when entering the name.

**Comment by DanThrane** [comment]
<img width="1616" alt="image" src="https://github.com/SDU-eScience/UCloud/assets/1782437/fdd8b02e-9b4c-4d5f-a966-a04524471efd">
Open terminal and open interface are placed incorrectly with multi-node jobs (I feel like you have mentioned a fix already, but just in case).

**Comment by Tiggles** [comment]
> It is not possible to create a folder when selecting an input folder for an app. The enter key does nothing when entering the name.

Already fixed

**Comment by Tiggles** [comment]
Job selector styling is off for column titles
<img width="852" alt="Screenshot 2024-03-13 at 08 58 57" src="https://github.com/SDU-eScience/UCloud/assets/7610257/8d929264-bb9e-49b9-992b-d2b2841ddd6e">

**Comment by MHvegholm** [comment]
![image](https://github.com/SDU-eScience/UCloud/assets/158156476/a92cf237-e6ce-41b0-bbc2-8cd1001ef3e0)
Workspace > usage leads to error screen, cannot escape without reloading (Crash?) – Only in my workspace
- When clicking on “Workspace” icon or accessing “workspace/usage” page while within “My workspace”, leads to error report page which cannot be left without reloading website ( Crash?)
- Error doesn’t occur when in workspace “Provider Type 1”

**Comment by MHvegholm** [comment]
![image](https://github.com/SDU-eScience/UCloud/assets/158156476/e431fbf7-dbfb-4a8c-b106-87559b688540)

Suggestion: addition of “jump-to workspace Rootdir/drive selection menu"
- It would be nice if the icon to the left of the displayed path was “click”-able and lead to the drive selection page 
- Alternative could be that a root/drive-dir before home (ex. “My workspace”), was shown and “click”-able, leading one back to the drive selection page.
- Would be equivalent to clicking on “files” icon, or “drives” title under files on the sidebar.

**Comment by MHvegholm** [comment]
> > Font color difficult to read in some dropdown menus:
> > ```
> > * Examples: job creation and grant application page
> > 
> > * Suggestion: Change to white
> > ```
> 
> Also not experiencing this. Which platform + browser are you using?

Chrome

**Comment by MHvegholm** [comment]
![image](https://github.com/SDU-eScience/UCloud/assets/158156476/c350f1eb-ed62-419f-85ad-8c21566d9032)
Suggestion: Embiggen “star” icon for “starring apps”

**Comment by Tiggles** [comment]
Favorite icon flicker on file browse component
https://github.com/SDU-eScience/UCloud/assets/7610257/77635ad4-37dd-40fa-b6f8-e6ad5b278eed

**Comment by DanThrane** [comment]
It should not be possible to select text in the flavor dropdown:

<img width="586" alt="image" src="https://github.com/SDU-eScience/UCloud/assets/1782437/90ccebf8-97d3-42b8-9444-4916435dd328">

Missing placeholder text for license servers:

<img width="829" alt="image" src="https://github.com/SDU-eScience/UCloud/assets/1782437/d8792b4f-5bb2-43f2-ab66-539caea51439">

Placeholder text is invisible on login page (compare this with production):

<img width="379" alt="image" src="https://github.com/SDU-eScience/UCloud/assets/1782437/a0b1a64e-1b75-4241-9eff-2bd4d504f3e8">

**Comment by Tiggles** [comment]
https://github.com/SDU-eScience/UCloud/assets/7610257/8872097f-f04f-4035-aa36-22f4bf76ddb8

**Comment by DanThrane** [comment]
Surprising behavior during drive creation:

1. Click 'Create drive'
2. Click somewhere else on the screen such that the input loses focus
3. This triggers a creation event which fails due to the title being blank

Generally this is the correct behavior, but I think we should check if the input field is empty. If the input field is empty then we should consider this a cancel instead of a create. If the input field is not empty, then we should continue treating it as a creation request.

**Comment by DanThrane** [comment]
![Uploading image.png…]()
Search query is not empty.

**Comment by JHavstein** [comment]
In the 'Files' menu, the buttons _Upload files_ and _Create folder_ have keyboard shortcuts shown in the button (picture 1=. These keyboard shortcuts are missing from the same two buttons from the job output page (picture 2). The keyboard shortcuts work in both places.
<img width="719" alt="Skærmbillede 2024-03-27 kl  13 25 51" src="https://github.com/SDU-eScience/UCloud/assets/33627730/d6b2f089-abf9-42ff-a634-3dc7e01427a0">
<img width="554" alt="Skærmbillede 2024-03-27 kl  13 26 09" src="https://github.com/SDU-eScience/UCloud/assets/33627730/e0884b0d-3938-49c4-8b45-f96105e52054">

**Comment by DanThrane** [comment]
"Launch with..." fails with a 400 bad request from the backend. This is a frontend issue since the wrong request type is being sent. There might still be backend issues once we begin sending the correct request type. 

<img width="925" alt="image" src="https://github.com/SDU-eScience/UCloud/assets/1782437/917fede3-e040-470d-9018-5c49dfce2133">

**Comment by Tiggles** [comment]
https://github.com/SDU-eScience/UCloud/assets/7610257/102e5095-5af1-4b09-95db-a670b2befed6

**Comment by DanThrane** [comment]
I think we should add an edge-case for jobs running with the unknown application (name = `unknown`). In this case it should probably skip the application name entirely and just display job name/id.

<img width="896" alt="image" src="https://github.com/SDU-eScience/UCloud/assets/1782437/177ccf34-b3e3-43a1-8c85-5568376082d5">

**Comment by Tiggles** [comment]
Not wide enough width for selector (on Chrome)
![image](https://github.com/SDU-eScience/UCloud/assets/7610257/8365a5e0-35d8-4b1b-9d11-5dfd3006ad70)

**Comment by DanThrane** [comment]
Drive rename field overlaps with provider icon 
<img width="668" alt="image" src="https://github.com/SDU-eScience/UCloud/assets/1782437/e39b89be-ad5b-4333-98de-9a337edd565a">

**Comment by Tiggles** [comment]
Delete Drive lost it's color
<img width="471" alt="Screenshot 2024-04-08 at 10 32 15" src="https://github.com/SDU-eScience/UCloud/assets/7610257/01c1af77-ba0b-4f30-b1db-4058d2e462df">

**Comment by Tiggles** [comment]
Drive-dropdown in FileBrowse is too wide
<img width="855" alt="Screenshot 2024-04-09 at 11 57 38" src="https://github.com/SDU-eScience/UCloud/assets/7610257/00abddae-9ccf-4f33-9d2f-3059bb9f8947">

**Comment by mrlhansen** [comment]
* Refreshing a page that ends with a `/` in the URL still results in a blank page - not sure if this is UI or not.
* The toogle for light/dark mode looks wrong.
* 
![image](https://github.com/SDU-eScience/UCloud/assets/31279460/3d5355bd-2ab2-43d4-a894-5a17d61fcaf4)

**Comment by xiroV** [comment]
A few comments (current dev state):

 - "Shared by" on "Invites" card on dashboard appears on two lines,
 - Rows on "Recent runs" card and "Grant applications" card on dashboard are selectable, but doesn't have any associated actions.
 - Header on files table (and resource tables) could have a text size of 100% instead of 110% (personal opinion).
 - As previously discussed (and also mentioned by Frederica), during product selection (during creation of resources, like drives), in case the user has access to resources from multiple providers, it isn't obvious which products belong to which provider. This is clear during machine selection before starting a job for example, but not in most other cases.

**Comment by DanThrane** [comment]
Markdown preview links are no longer correct due to changes in preview URL.

**Comment by Tiggles** [comment]
> Markdown preview links are no longer correct due to changes in preview URL.

I think I need some more details
<img width="794" alt="Screenshot 2024-04-16 at 09 34 27" src="https://github.com/SDU-eScience/UCloud/assets/7610257/5b0ce1c8-325f-4683-ac27-cbcc224e9aaa">

**Comment by DanThrane** [comment]
Sizing of console is slightly wrong (low priority in my opnion)

![image](https://github.com/SDU-eScience/UCloud/assets/1782437/e1bc7784-09ba-487c-9971-af2290717a8f)

**Comment by mrlhansen** [comment]
The issue with URLs ending in a `/` also causes the link to `files` to not work here (also, should that really open in a new window?).
![image](https://github.com/SDU-eScience/UCloud/assets/31279460/1173acf8-944b-4433-856b-2f5cdbe11fc5)

**Comment by Tiggles** [comment]
> (also, should that really open in a new window?)

Yes, otherwise the form/arguments are wiped when returning to the page.

**Comment by DanThrane** [comment]
Browsing files causes `GET /api/jobs/retrieveProducts?providers=$PROVIDER_ID` and `GET /api/iapps/syncthing/retrieve?provider=$PROVIDER_ID&productId=syncthing` to be invoked on every folder instead of being cached.

**Comment by mrlhansen** [comment]
On the SSH keys page (https://dev.cloud.sdu.dk/app/ssh-keys).

* The button "Create SSH key" is maybe a bit misnamed. I would rename it to "Add" or "Upload" instead of "Create".
* When adding a SSH key (https://dev.cloud.sdu.dk/app/ssh-keys/create) the title "SSH keys" is also off to the left. Maybe the title should also be renamed to e.g. "New SSH key". The title text is also quite small (?) but I am not sure if that is just the case in general.

**Comment by DanThrane** [comment]
Drag-to-select no longer works if you are starting the drag from the "outside". That is not on a row.

**Comment by Tiggles** [comment]
Did it work before, dragging from the outside in?

**Comment by Tiggles** [comment]
Next #4177
